android/services/mongodb-remote/src/androidTest/java/com/mongodb/stitch/android/services/mongodb/remote/internal/SyncPerformanceIntTestsHarness.kt

```diff
@@ -0,0 +1,421 @@
+
+    // Private constants
+    private val mongodbUriProp = "test.stitch.mongodbURI"
+    private val stitchAPIKeyProp = "test.stitch.androidPerfStitchAPIKey"
+
+    private val stitchOutputAppName = "android-sdk-perf-testing-yuvef"
+    private val stitchOutputDbName = "performance"
+    private val stitchOutputCollName = "results"
+    private var stitchTestHost = ""
+
+    // Private variables
+    private lateinit var outputClient: StitchAppClient
+    private lateinit var outputColl: RemoteMongoCollection<Document>
+    private lateinit var mdbService: Apps.App.Services.Service
+    private lateinit var mdbRule: RuleResponse
+
+    // Public variables
+    lateinit var testClient: StitchAppClient
+    var testDbName = ObjectId().toHexString()
+    var testCollName = ObjectId().toHexString()
+    var testNamespace = MongoNamespace(testDbName, testCollName)
+    lateinit var testMongoClient: RemoteMongoClient
+    lateinit var testColl: RemoteMongoCollection<Document>
+    lateinit var testUserId: String
+
+    val testDataSynchronizer: DataSynchronizer
+        get() = (testMongoClient as RemoteMongoClientImpl).dataSynchronizer
+    val testNetworkMonitor: BaseStitchIntTest.TestNetworkMonitor
+        get() = BaseStitchAndroidIntTest.testNetworkMonitor
+
+    private fun getStitchAPIKey(): String {
+        return InstrumentationRegistry.getArguments().getString(stitchAPIKeyProp, "")
+    }
+
+    /**
+     * Get the uri for where mongodb is running locally.
+     */
+    private fun getMongoDbUri(): String {
+        return InstrumentationRegistry.getArguments().getString(mongodbUriProp, "mongodb://localhost:26000")
+    }
+
+    override fun getStitchBaseURL(): String {
+        if (stitchTestHost.isNotEmpty()) {
+            return stitchTestHost
+        }
+        return InstrumentationRegistry.getArguments().getString(
+            "test.stitch.baseURL",
+            "http://10.0.2.2:9090"
+        )
+    }
+
+    private val transport = OkHttpInstrumentedTransport()
+    override fun getAppClientConfigurationBuilder(): StitchAppClientConfiguration.Builder {
+        return super.getAppClientConfigurationBuilder().withTransport(transport)
+    }
+
+    @Before
+    override fun setup() {
+        super.setup()
+
+        if (!Stitch.hasAppClient(stitchOutputAppName)) {
+            outputClient = Stitch.initializeAppClient(stitchOutputAppName)
+        } else {
+            outputClient = Stitch.getAppClient(stitchOutputAppName)
+        }
+
+        if (!outputClient.auth.isLoggedIn) {
+            Tasks.await(outputClient.auth.loginWithCredential(UserApiKeyCredential(getStitchAPIKey())))
+        }
+
+        outputColl = outputClient
+            .getServiceClient(RemoteMongoClient.factory, "mongodb-atlas")
+            .getDatabase(stitchOutputDbName)
+            .getCollection(stitchOutputCollName)
+    }
+
+    private fun setupIter() {
+
+        // Setup Stitch app to use for the tests
```

Comment:
[minor] I think this is fine for now, but long term as we move to testing against prod stitch, we won't be able to create new apps on the fly unless we simulate mongodb-cloud auth and log in via the admin SDK.

We might want to come up with setup and teardown methods that reuse the same stitch app if we're testing against prod. Whether or not we do that now is up to you. I think it's fine to leave as is, and file a ticket to do the work for testing against prod later, and we can just do Swift with both ways from the get-go